src/main/java/su/litvak/chromecast/api/v2/MediaMetadata.java

```diff
@@ -0,0 +1,246 @@
+package su.litvak.chromecast.api.v2;
+
+import org.codehaus.jackson.annotate.JsonAnySetter;
+import org.codehaus.jackson.annotate.JsonProperty;
+import org.codehaus.jackson.annotate.JsonSubTypes;
+import org.codehaus.jackson.annotate.JsonTypeInfo;
+import org.codehaus.jackson.map.annotate.JsonSerialize;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "metadataType")
+@JsonSubTypes({
```

Comment:
I was able to figure out how to create instances of the right type via the `metadataType` property, but I can't get the instances to be populated with a Map. So using `JsonAnySetter` as a workaround.